cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -96,19 +97,40 @@ func (c *CStorPoolController) syncHandler(key string, operation common.QueueOper
 	return nil
 }
 
+// cStorPoolAddEventHandler calls cStorPoolAddEvent, and makes in-mem structures of successfully imported/created pool
+func (c *CStorPoolController) cStorPoolAddEventHandler(cStorPoolGot *apis.CStorPool) (string, error) {
+	var zpoolDumpErr error
+	pool.RunnerVar = util.RealRunner{}
+	uid := string(cStorPoolGot.GetUID())
+	csp := pool.ImportedCStorPools[uid]
+
+	common.SyncResources.Mux.Lock()
+	if csp != nil {
+		c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.AlreadyPresent), string(common.MessageResourceAlreadyPresent))
+	}
+	status, err := c.cStorPoolAddEvent(cStorPoolGot)
+	if status == string(apis.CStorPoolStatusOnline) {
+		pool.ImportedCStorPools[uid] = cStorPoolGot.DeepCopy()
+		pool.CStorZpools[uid], zpoolDumpErr = zpool.Dump()
+		if zpoolDumpErr != nil {
+			glog.Errorf("failed in getting zpool dump %v", zpoolDumpErr)
+			delete(pool.CStorZpools, uid)
```

Comment:
i guess this runs in a loop, is it possible that pool.CStorZpools is modifies at the same time in other goroutine or it is empty (i see we are initializing with empty map) ? If yes, should we check whether uid is present or not in the map before deletion ?